src/EFCore.PG/Query/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -206,6 +206,86 @@ protected override Expression VisitSqlBinary(SqlBinaryExpression binary)
         }
     }
 
+    protected override Expression VisitDelete(DeleteExpression deleteExpression)
```

Comment:
@smitpatel here's the PG join logic for USING, let me know if you have any thoughts.

One thing that crossed my mind... This may be a bit of "translation" logic embedded in the query SQL generation, e.g. moving the join condition out to the predicate. We could consider having a provider-specific PostgresDeleteExpression, which corresponds exactly to the PG SQL (target table, optional USINGs, predicate); a post-processing visitor would then convert the relational expression (which is a thin wrapper over SelectExpression) into that thing. This would keep SQL generation very simple, only rendering into SQL. But I'm not sure it's worth it, things look OK as-is too. Let me know what you think.